lavalink/nodemanager.py

```diff
@@ -100,9 +116,24 @@ def find_ideal_node(self, region: str = None):
         best_node = self.find_ideal_node(node.region)
 
         if not best_node:
-            # TODO: Move players to a queue, wait for a node to connect?
-            log.error('Unable to move players, no available nodes!')
+            for player in node.players:
+                self.player_queue.append(player)
+
+            await self._wait_for_best_node(node)
+            log.error('Unable to move players, no available nodes! Trying to find one to connect to.')
             return
 
         for player in node.players:
             await player.change_node(best_node)
+
+    async def _wait_for_best_node(self, original_node):
```

Comment:
I do not think using a `while True` is a good idea, IMO, it would be better by handling the players reconnection in  `_node_connect`  as the thing you're waiting for is to a node to connect.

Reply:
agreed, you should never use `while True` it could cause a deadlock and you won't be able to stop the current process normally, you'd have to kill it. I'd suggest doing a stripped down version of while loop in `websockets.py` with a variable backoff.

Reply:
The best solution IMO is to use `node_connected` event in `node_manager.py` and shift all players (if any) to the node that has just connected to the client. Maybe perhaps move players to a closer region as another node becomes available.